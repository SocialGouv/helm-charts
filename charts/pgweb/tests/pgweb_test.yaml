suite: test pgweb
#templates:
tests:
  - it: should render default
    asserts:
      - template: "deployment.yaml"
        matchSnapshot:
          path: spec
      - template: "ingress.yaml"
        hasDocuments:
          count: 0
      - template: "service.yaml"
        matchSnapshot:
          path: spec
      - template: "hpa.yaml"
        hasDocuments:
          count: 0
      - template: "serviceaccount.yaml"
        matchSnapshot:
          path: metadata
  - it: should render with HPA
    values:
      - values-hpa.yaml
    asserts:
      - template: "deployment.yaml"
        matchSnapshot:
          path: spec
      - template: "ingress.yaml"
        hasDocuments:
          count: 0
      - template: "service.yaml"
        matchSnapshot:
          path: spec
      - template: "hpa.yaml"
        hasDocuments:
          count: 1
      - template: "hpa.yaml"
        matchSnapshot:
          path: spec
      - template: "serviceaccount.yaml"
        matchSnapshot:
          path: metadata
  - it: should render with ingress
    values:
      - values-ingress.yaml
    asserts:
      - template: "ingress.yaml"
        matchSnapshot:
          path: spec
      - template: "ingress.yaml"
        hasDocuments:
          count: 1
