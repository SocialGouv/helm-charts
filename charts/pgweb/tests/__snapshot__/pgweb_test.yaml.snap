should render default:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: pgweb
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: pgweb
      spec:
        containers:
          - image: sosedoff/pgweb:0.14.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: http
            name: pgweb
            ports:
              - containerPort: 8081
                name: http
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /
                port: http
            resources: {}
            securityContext: {}
        securityContext: {}
        serviceAccountName: RELEASE-NAME-pgweb
  2: |
    ports:
      - name: http
        port: 80
        protocol: TCP
        targetPort: http
    selector:
      app.kubernetes.io/instance: RELEASE-NAME
      app.kubernetes.io/name: pgweb
    type: ClusterIP
  3: |
    labels:
      app.kubernetes.io/instance: RELEASE-NAME
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: pgweb
      helm.sh/chart: pgweb-1.9.0
    name: RELEASE-NAME-pgweb
should render with HPA:
  1: |
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: pgweb
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: pgweb
      spec:
        containers:
          - image: sosedoff/pgweb:0.14.1
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: http
            name: pgweb
            ports:
              - containerPort: 8081
                name: http
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /
                port: http
            resources: {}
            securityContext: {}
        securityContext: {}
        serviceAccountName: RELEASE-NAME-pgweb
  2: |
    ports:
      - name: http
        port: 80
        protocol: TCP
        targetPort: http
    selector:
      app.kubernetes.io/instance: RELEASE-NAME
      app.kubernetes.io/name: pgweb
    type: ClusterIP
  3: |
    maxReplicas: 100
    metrics:
      - resource:
          name: cpu
          targetAverageUtilization: 80
        type: Resource
    minReplicas: 1
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: RELEASE-NAME-pgweb
  4: |
    labels:
      app.kubernetes.io/instance: RELEASE-NAME
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: pgweb
      helm.sh/chart: pgweb-1.9.0
    name: RELEASE-NAME-pgweb
should render with ingress:
  1: |
    rules:
      - host: chart-example.local
        http:
          paths:
            - backend:
                service:
                  name: RELEASE-NAME-pgweb
                  port:
                    number: 80
              path: /
              pathType: ImplementationSpecific
